test/SILOptimizer/opt-remark-generator.swift

```diff
@@ -3,71 +3,75 @@
-// CHECK-NEXT: Args:
-// CHECK-NEXT:   - String:          'Found release:'
-// CHECK-NEXT:   - InferValueFailure: Unable to infer any values being released.
-// CHECK-NEXT: ...
-// CHECK-NEXT: --- !Missed
-// CHECK-NEXT: Pass:            sil-opt-remark-gen
-// CHECK-NEXT: Name:            sil.memory-management
-// CHECK-NEXT: DebugLoc:        { File: '{{.*}}opt-remark-generator.swift', 
-// CHECK-NEXT:                    Line: 67, Column: 12 }
-// CHECK-NEXT: Function:        'useGlobal()'
-// CHECK-NEXT: Args:
-// CHECK-NEXT:   - String:          'Found release:'
-// CHECK-NEXT:   - InferredValue:   'on value:'
-// CHECK-NEXT:     DebugLoc:        { File: '{{.*}}opt-remark-generator.swift',
-// CHECK-NEXT:                        Line: 64, Column: 9 }
-// CHECK-NEXT: ...
-
 public class Klass {}
 
 public var global = Klass()
 
 @inline(never)
 public func getGlobal() -> Klass {
+
+    // CHECK: --- !Missed
+    // CHECK-NEXT: Pass:            sil-opt-remark-gen
+    // CHECK-NEXT: Name:            sil.memory-management
+    // CHECK-NEXT: DebugLoc:        { File: '{{.*}}opt-remark-generator.swift',
+    // CHECK-NEXT:                    Line: [[#@LINE+8]], Column: 5 }
+    // CHECK-NEXT: Function:        'getGlobal()'
+    // CHECK-NEXT: Args:
+    // CHECK-NEXT:   - String:          'Found retain:'
+    // CHECK-NEXT:   - InferredValue:   'on value:'
+    // CHECK-NEXT:     DebugLoc:        { File: '{{.*}}opt-remark-generator.swift',
+    // CHECK-NEXT:                        Line: [[#@LINE-15]], Column: 12 }
+    // CHECK-NEXT: ...
     return global // expected-remark @:5 {{Found retain:}}
-                  // expected-note @-5:12 {{on value:}}
+                  // expected-note @-18:12 {{on value:}}
 }
 
 public func useGlobal() {
     let x = getGlobal()
+
+    // CHECK-NEXT: --- !Missed
+    // CHECK-NEXT: Pass:            sil-opt-remark-gen
+    // CHECK-NEXT: Name:            sil.memory-management
+    // CHECK-NEXT: DebugLoc:        { File: '{{.*}}opt-remark-generator.swift',
+    // CHECK-NEXT:                    Line: [[# @LINE + 35]], Column: 5 }
+    // CHECK-NEXT: Function:        'useGlobal()'
+    // CHECK-NEXT: Args:
+    // CHECK-NEXT:   - String:          'Found retain:'
+    // CHECK-NEXT:   - InferredValue:   'on value:'
+    // CHECK-NEXT:     DebugLoc:        { File: '{{.*}}opt-remark-generator.swift',
+    // CHECK-NEXT:                        Line: [[# @LINE - 12]], Column: 9 }
+    // CHECK-NEXT: ...
+    // CHECK-NEXT: --- !Missed
+    // CHECK-NEXT: Pass:            sil-opt-remark-gen
+    // CHECK-NEXT: Name:            sil.memory-management
+    // CHECK-NEXT: DebugLoc:        { File: '{{.*}}opt-remark-generator.swift',
+    // CHECK-NEXT:                    Line: [[# @LINE + 23]], Column: 12 }
+    // CHECK-NEXT: Function:        'useGlobal()'
+    // CHECK-NEXT: Args:
+    // CHECK-NEXT:   - String:          'Found release:'
+    // CHECK-NEXT:   - InferredValue:   'on variadic argument array for argument 0:'
```

Comment:
I fixed up some tests here to make these tests easier to maintain by using @LINE, etc, but substantively this is the test that shows that we properly identified the variadic argument array.